sanitizers/detectors/path_traversal.go

```diff
@@ -6,32 +6,35 @@ import (
 	"strings"
 
 	"github.com/CodeIntelligenceTesting/gofuzz/sanitizers/fuzzer"
+	"github.com/CodeIntelligenceTesting/gofuzz/sanitizers/reporter"
 )
 
 var (
-	dotDotPath         = filepath.Join("..", "..", "evil_path")
-	PathTraversalError = errors.New("path traversal")
+	dotDotPath         string
+	PathTraversalError = errors.New("Path traversal error")
 )
 
-type PathTraversal struct {
-	id   int    // numeric identifier to distinguish between the detectors for the various call sites
-	path string // path passed to the file API function
+func init() {
+	dotDotPath = filepath.Join("..", "..", "evil_path")
 }
 
-// Make sure that the path traversal detector implements the Detector interface
-var _ Detector = (*PathTraversal)(nil)
+var _ Detector = (*pathTraversal)(nil)
 
-func (pt *PathTraversal) Detect() error {
-	if strings.Contains(pt.path, dotDotPath) {
-		return PathTraversalError
-	}
-	fuzzer.GuideTowardsContainment(pt.path, dotDotPath, pt.id)
-	return nil
+type pathTraversal struct {
+	id  int
+	cmd string
 }
 
-func NewPathTraversal(id int, path string) *PathTraversal {
-	return &PathTraversal{
-		id:   id,
-		path: path,
+var PathTraversal = pathTraversal{}
+
+func (pt *pathTraversal) Detect() {
+	if strings.Contains(pt.cmd, dotDotPath) {
+		reporter.ReportFindingf("%s: vulnerable path %q", PathTraversalError.Error(), pt.cmd)
+		return
 	}
+	fuzzer.GuideTowardsContainment(pt.cmd, dotDotPath, pt.id)
+}
+
+func (pt *pathTraversal) New(id int, cmd interface{}) *pathTraversal {
```

Comment:
Same as above regarding the parameter of `interface{}` type

Reply:
same reasoning as earlier with `GetCommand`